packages/testing/src/index.ts

```diff
@@ -387,6 +407,8 @@ export const defaultActivityInfo: activity.Info = {
   currentAttemptScheduledTimestampMs: 1,
 };
 
+const mockActivityContextStorage: AsyncLocalStorage<() => Promise<unknown>> = new AsyncLocalStorage();
```

Comment:
I like the idea of code reuse here but IMHO it's not worth the complication.
If all you want is to inject interceptors, I'd add that functionality here.

Reply:
It's also to get proper lifecycle event logging, catching of cancellation error, correct stack trace truncation...

Reply:
It should however be possible to avoid this `AyncLocalStorage` by passing the activity function to `Activity.runNoEncoding()`, and from there to `Activity.execute()`... Let me see if that turns out better.